### DISCUSSION TITLE
AI animation

### DISCUSSION THREAD

**Discussion by josplit** [continue-revolution/sd-webui-segment-anything#76] [General]
Hello! I'm mr boofy (on twitter) I do animations with stable diffusion, I've always thought that SAM and DINO are the key to more stable animations and more drastic changes, do you think that with your scripts I can use it to batch render each frame and that in each frame Can you identify what I'm asking for consistently or is it only possible with one image? Installing to test! thank you!

**Comment by continue-revolution** [comment]
The short answer is yes, you may do batch segmentation with either SAM+DINO (bounding box -> segmentation) or SAM+Oneformer (enhanced semantic segmentation, under AutoSAM/Mask by Category/Batch process tab). 

The long answer is to read README. Read [GroundingDINO](https://github.com/continue-revolution/sd-webui-segment-anything#groundingdino), [AutoSAM](https://github.com/continue-revolution/sd-webui-segment-anything#autosam), [How to Use](https://github.com/continue-revolution/sd-webui-segment-anything#how-to-use), [Demo (image demos are enough for your need)](https://github.com/continue-revolution/sd-webui-segment-anything#demo
) for instructions on how to do it.

As far as I know, someone has already done so on [bilibili](https://www.bilibili.com/video/BV1cX4y1z7Cb).
